renderer/brave_content_renderer_client.cc

```diff
@@ -185,6 +199,18 @@ void BraveContentRendererClient::RenderFrameCreated(
         ISOLATED_WORLD_ID_BRAVE_INTERNAL);
   }
 #endif
+
+  if (ai_chat::features::IsAIChatEnabled() && !IsIncognitoProcess()) {
```

Comment:
Also related (not for this PR) checking `!IsIncognitoProcess()` is not ideal here. It would be better to have a helper method in common that takes a bool for incognito so you can use the same check internally on the browser and renderer side (would be called inside ai_chat::IsAllowedForContext). Also the pref should be passed through `BraveRendererUpdater`. `ai_chat::IsAllowedForContext`/`ai_chat::IsAIChatEnabled`/`ai_chat::features::IsAIChatEnabled` also gets confusing and not sure how `HasUserOptedIn` fits in to that.